services/configuration/src/schema.rs

```diff
@@ -52,15 +52,70 @@ lazy_static! {
         &Hash::zero(),
         &Signature::zero(),
     ).into_bytes();
+    static ref VOTE_DECISION_BYTES_MARKER: Vec<u8> = b"consent".to_vec();
+    static ref VOTE_AGAINST_DECISION_BYTES_MARKER: Vec<u8> = b"dissent".to_vec();
 }
 
-/// A functional equivalent to `Option<Vote>` used to store votes in the service schema.
+/// A enum used to represent different kinds of vote, `Vote` and `VoteAgainst` transactions.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum VotingDecision {
+    /// `Vote` transaction variant
+    Vote(Vote),
+    /// `VoteAgainst` transaction variant
+    VoteAgainst(VoteAgainst),
+}
+
+impl CryptoHash for VotingDecision {
+    fn hash(&self) -> Hash {
+        match *self {
+            VotingDecision::Vote(ref vote) => vote.hash(),
+            VotingDecision::VoteAgainst(ref vote_against) => vote_against.hash(),
+        }
+    }
+}
+
+trait VotingDecisionBytes: StorageValue {
+    fn decision_marker() -> Vec<u8>;
+
+    fn has_decision_marker(bytes: &Cow<[u8]>) -> bool {
+        let marker = Self::decision_marker();
+        let idx = marker.len();
+        let (marker_bytes, _tx_bytes) = bytes.split_at(idx);
+        marker.as_slice().eq(marker_bytes)
+    }
+
+    fn into_decision_bytes(self) -> Vec<u8> {
+        let mut decision_bytes = Self::decision_marker();
+        decision_bytes.extend(self.into_bytes());
+        decision_bytes
+    }
+
+    fn from_decision_bytes(bytes: Cow<[u8]>) -> Self {
+        let idx = Self::decision_marker().len();
+        let (_marker_bytes, tx_bytes) = bytes.split_at(idx);
+        Self::from_bytes(Cow::from(tx_bytes))
+    }
+}
+
+impl VotingDecisionBytes for Vote {
+    fn decision_marker() -> Vec<u8> {
+        VOTE_DECISION_BYTES_MARKER.clone()
+    }
+}
+
+impl VotingDecisionBytes for VoteAgainst {
+    fn decision_marker() -> Vec<u8> {
+        VOTE_AGAINST_DECISION_BYTES_MARKER.clone()
+    }
+}
+
+/// A functional equivalent to `Option<T>` used to store votes in the service schema.
 ///
 /// # Notes
 ///
-/// The `None` variant of the type is represented by a `Vote` with all bytes set to zero.
+/// The `None` variant of the type is represented by a `T` with all bytes set to zero.
```

Comment:
This change is incorrect; the previous statement ("...by a `Vote`...") remains correct.

Reply:
Done.